### PR TITLE
Fix #691: Landscape Lowfi Reset PIN

### LINKED ISSUE #691: Landscape Lowfi: Reset PIN
Mock: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/6f83fcd9-e72d-4941-a16f-6754a869e083/RG-MN-Register-Empty-37

Write tests too.

### PR DESCRIPTION
**Pull request by gunishjain**
## Explanation
Created Landscape View for Reset Pin and wrote test cases.
  

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by gunishjain** [comment]
@rt4914 As suggested by you, I copy-pasted initial code from portrait and then proceeded with the modifications. Test cases are written and tested accordingly.

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -70,8 +70,8 @@
     <activity
       android:name=".settings.profile.ProfileRenameActivity" />
     <activity
-      android:name=".settings.profile.ProfileResetPinActivity"
-      android:screenOrientation="portrait" />
+      android:name=".settings.profile.ProfileResetPinActivity">
+    </activity>
```
Comment (SayantanBanerjee16) [inline app/src/main/AndroidManifest.xml]: Refactor this as follows :-

`<activity`
`      android:name=".settings.profile.ProfileResetPinActivity" />`

File: app/src/sharedTest/java/org/oppia/app/settings/profile/ProfileResetPinActivityTest.kt [hunk 2]
```diff
@@ -165,6 +168,17 @@ class ProfileResetPinActivityTest {
       onView(allOf(withId(R.id.error_text), isDescendantOfA(withId(R.id.input_confirm_pin)))).check(matches(withText("")))
     }
   }
+  @Test
+  fun testProfileResetPinActivity_startActivitywithAdmin_configurationLandscape_isCorrect() {
+    ActivityScenario.launch<ProfileResetPinActivity>(ProfileResetPinActivity.createProfileResetPinActivity(context, 0, true)).use{
+      onView(ViewMatchers.isRoot()).perform(orientationLandscape())
```
Comment (SayantanBanerjee16) [inline app/src/sharedTest/java/org/oppia/app/settings/profile/ProfileResetPinActivityTest.kt]: Refractor it as below to make better readability:

`onView(isRoot()).perform(orientationLandscape())`

and subsequently imports will change and thus delete unused imports

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (rt4914) [inline app/src/main/AndroidManifest.xml]: Nit update:
`<activity android:name=".settings.profile.ProfileResetPinActivity" />`

File: app/src/sharedTest/java/org/oppia/app/settings/profile/ProfileResetPinActivityTest.kt [hunk 3]
```diff
@@ -165,6 +168,17 @@ class ProfileResetPinActivityTest {
       onView(allOf(withId(R.id.error_text), isDescendantOfA(withId(R.id.input_confirm_pin)))).check(matches(withText("")))
     }
   }
+  @Test
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/settings/profile/ProfileResetPinActivityTest.kt:172]: Add one blank line above this.

**Review by SayantanBanerjee16 — CHANGES_REQUESTED** [review]
@gunishjain Please take a look at some of the NIT changes. Thanks :)

**Review by rt4914 — CHANGES_REQUESTED** [review]
Three different cases of reset pin.
1. Type pin in `first-edit-text` -> Rotate -> The pin should still be there. -> **NOT WORKING**
2. Type pin in `first-edit-text` & `second-edit-text` -> Rotate -> The pin should still be there. -> Working.
3. Type pin in `second-edit-text` -> Rotate -> The pin should still be there. -> **NOT WORKING**

You need to fix the 1. and 3. case in the corresponding Kotlin files and later on we will add test cases around this too.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (gunishjain) [reply app/src/main/AndroidManifest.xml]: @rt4914 You said to fix 1 and 3, but told that we will add test cases later, so should I check previously written test cases?

Comment (rt4914) [reply app/src/main/AndroidManifest.xml]: No actually, you should follow the steps mentioned in 1 and 3. Our final aim is to make sure that if user has typed something and does orientation change in between than that text should not get removed.

Comment (gunishjain) [reply app/src/main/AndroidManifest.xml]: @rt4914 Not able to figure it out, What I am thinking is that I need to look at the view model for profile reset pin activity/presenter and make some changes for a configuration change. Can you provide me a hint?

Comment (rt4914) [reply app/src/main/AndroidManifest.xml]: @gunishjain You need to make sure that if user enter's some content before doing orientation change, then after orientation change that content should be present in the edit text.

**Comment by rt4914** [comment]
@gunishjain Any updates on this PR?

**Comment by gunishjain** [comment]
> @gunishjain Any updates on this PR?

**Comment by gunishjain** [comment]
@rt4914 Not able to solve case 1 and 3.

**Comment by rt4914** [comment]
@gunishjain You had mentioned that you have already finished the test cases but I am not able to see any test cases that I had mentioned above. Check `Files Changed` section in this PR and you will see what I am talking about.

**Review by rt4914 — CHANGES_REQUESTED** [review]
In your test cases you are just check whether the input_pin is displayed or not. But you are not checking whether the string that you entered before orientation change is there or not.
Also, the implementation that I had mentioned earlier, meaning that when you do the orientation change the pins in edittext should persist, that implementation is still not there. 

Also, I have come to know after this code that you have just written the code but you have not tested the code by running the app and playing with the app. 
Please make sure that you understand the issue first.

**Comment by rt4914** [comment]
Closing this PR as the author won't be able to contribute on this.

**Pull request closed without merging**